public/sites/default/settings.php

```diff
@@ -182,34 +168,66 @@ function drupal_get_env(string|array $variables) : mixed {
 $config['helfi_api_base.environment_resolver.settings']['environment_name'] = getenv('APP_ENV');
 $config['helfi_api_base.environment_resolver.settings']['project_name'] = getenv('PROJECT_NAME');
 
-// settings.php doesn't know about existing configuration yet so we can't
-// just append new headers to an already existing headers array here.
-// If you have configured any extra headers in your purge settings
-// you must add them in your all.settings.php as well.
-// @todo Replace this with config override service?
-$config['varnish_purger.settings.default']['headers'] = [
-  [
-    'field' => 'Cache-Tags',
-    'value' => '[invalidation:expression]',
-  ],
-];
+if ($varnish_host = getenv('DRUPAL_VARNISH_HOST')) {
+  // Cache everything for 1 year by default.
+  $config['system.performance']['cache']['page']['max_age'] = 31536000;
 
-$config['varnish_purger.settings.varnish_purge_all']['headers'] = [
-  [
-    'field' => 'X-VC-Purge-Method',
-    'value' => 'regex',
-  ],
-];
+  $varnish_backend = parse_url($drush_options_uri, PHP_URL_HOST);
 
-if ($varnish_purge_key = getenv('VARNISH_PURGE_KEY')) {
-  $config['varnish_purger.settings.default']['headers'][] = [
-    'field' => 'X-VC-Purge-Key',
-    'value' => $varnish_purge_key,
-  ];
-  $config['varnish_purger.settings.varnish_purge_all']['headers'][] = [
-    'field' => 'X-VC-Purge-Key',
-    'value' => $varnish_purge_key,
+  if (getenv('APP_ENV') === 'local') {
+    // Varnish backend is something like varnish-helfi-kymp.docker.so on
+    // local env.
+    $varnish_backend = 'varnish-' . $varnish_backend;
+  }
+
+  // settings.php doesn't know about existing configuration yet so we can't
+  // just append new headers to an already existing headers array here.
+  // If you have configured any extra headers in your purge settings
+  // you must add them in your all.settings.php as well.
+  // @todo Replace this with config override service?
+  $varnishConfiguration = [
+    'default' => [
+      [
+        'field' => 'Cache-Tags',
+        'value' => '[invalidation:expression]',
+      ],
+    ],
+    'assets' => [
+      [
+        'field' => 'X-VC-Purge-Method',
+        'value' => 'regex',
+      ],
+      [
+        'field' => 'Host',
+        'value' => $varnish_backend,
+      ],
+    ],
+    'varnish_purge_all' => [
+      [
+        'field' => 'X-VC-Purge-Method',
+        'value' => 'regex',
+      ],
+    ],
   ];
+
+  foreach ($varnishConfiguration as $name => $headers) {
+    $config['varnish_purger.settings.' . $name]['hostname'] = $varnish_host;
+
+    if ($varnish_port = getenv('DRUPAL_VARNISH_PORT')) {
+      $config['varnish_purger.settings.' . $name]['port'] = $varnish_port;
+    }
+  }
+
+  foreach ($headers as $header) {
+    $config['varnish_purger.settings.' . $name]['headers'][] = $header;
+  }
```

Comment:
```diff
diff --git a/public/sites/default/settings.php b/public/sites/default/settings.php
index 6683510..fadca81 100755
--- a/public/sites/default/settings.php
+++ b/public/sites/default/settings.php
@@ -216,10 +216,10 @@ function drupal_get_env(string|array $variables) : mixed {
     if ($varnish_port = getenv('DRUPAL_VARNISH_PORT')) {
       $config['varnish_purger.settings.' . $name]['port'] = $varnish_port;
     }
-  }
 
-  foreach ($headers as $header) {
-    $config['varnish_purger.settings.' . $name]['headers'][] = $header;
+    foreach ($headers as $header) {
+      $config['varnish_purger.settings.' . $name]['headers'][] = $header;
+    }
   }
 
   if ($varnish_purge_key = getenv('VARNISH_PURGE_KEY')) {
```

Reply:
Yes it should. Thanks!

Reply:
Setting the purge key should be in the loop as well.

Reply:
I don't think so. It has to be set to every `varnish_purger.settings.*` item regardless of what's defined in `$varnishConfiguration`

Reply:
Ah, which is `varnish_purge_all`. That makes it to apply to all?

Reply:
No. I'm just being dumb :see_no_evil:  Should be fixed now